libr/core/cmd_open.c

```diff
@@ -1216,6 +1225,76 @@ static bool desc_list_json_cb(void *user, void *data, ut32 id) {
 	return true;
 }
 
+typedef struct {
+	int fd; // input and output
+	int inc;
+	int prevfd;
+	bool picknext;
+	bool found;
+	RCore *core;
+} RCmdOpData;
+
+static bool try_next(void *_user, void *_data, ut32 id) {
+	RCmdOpData *data = _user;
+	if (data->picknext) {
```

Comment:
having all that RCmdOpData for a simple operation such as this one looks too much to me. Getting the next/previous desc is really as simple as: 1) find the current descriptor in the list 2) get the next/prev descriptor. This `try_next` make it look much harder than what it is IMHO.

Btw, it seems `picknext` is used somehow in a unintuitive way. In this line, if data->picknext is true, you don't pick any next thing, you just set the fd and pick it, so it does the opposite than what the name suggests.